vignettes/GMAT-GMST.Rmd

```diff
@@ -0,0 +1,126 @@
+
+This vignette demonstrates how to set up the calculation between GMAT and GMST and how to plot results to observe differences over time. 
+
+## Setup
+
+Let's load the packages we will need and run a scenario.
+
+``` {r hector.init}
+library(hector)
+library(ggplot2)
+
+inifile <- file.path(system.file('input', package = 'hector'),
+                     'hector_rcp45.ini')
+core <- newcore(inifile)
+run(core)
+```
+
+
+## Find GMST
+
+GMST can be calculated from Hector outputs by computing a weighted sum of the land air and ocean surface temperatures using the following equation:
+
+$ GMST = (land air temperature * fractional land area) + (ocean surface temperature * (1 - fractional land area)) $
+
+In Hector, land air temperature is represented by `LAND_AIR_TEMP()` and `Tgav_land` while ocean surface temperature by `OCEAN_SURFACE_TEMP()` and `Tgav_ocean_ST`. These components stem from the variable `Tgav`, which is how Hector stores GMAT (along with `GLOBAL_TEMP()`.)
+
+``` {r find.GMST}
+flnd = 0.29;     # fractional land area, a constant in Hector
+
+hector_output <- fetchvars(core, 1745:2300, vars = c(LAND_AIR_TEMP(), OCEAN_SURFACE_TEMP(), GLOBAL_TEMP()))
+
+# Get the variables we want and corresponding years/value
+output <- subset(hector_output, variable %in% c(LAND_AIR_TEMP(), OCEAN_SURFACE_TEMP()))
+output <- subset(output, select = c(year, variable, value))
+
+# Reorder data
+output <- reshape(output, direction = "wide", idvar = "year", timevar = "variable")
+names(output) <- gsub("value.", "", names(output))
+
+# Conversion equation
```

Comment:
This is OK, although it seems simper to do something like
```r
output$GMST <- with(output, (Tgav_land * flnd + Tgav_ocean_ST * (1-flnd)) / 2)
```

(Haven't tested this on the command line but seems like it should work?)

Reply:
That would work but don't divide by 2.